File: pulumi/python/vault/mini/policies.py
MAGISTRATE = '''
path "*" {
  capabilities = ["create", "read", "update", "delete", "list", "sudo"]
}
path "sys/leases/lookup" {
  capabilities = ["list", "sudo"]
}
'''

TEMPLATE_READ = '''
path "secret/data/{group}/*" {{
  capabilities = ["read","list"]
}}
path "secret/metadata/{group}/*" {{
  capabilities = ["read","list"]
}}
'''

TEMPLATE_ADMIN = '''
path "secret/data/{group}/*" {{
  capabilities = ["create","read","update","delete","list"]
}}
path "secret/metadata/{group}/*" {{
  capabilities = ["read","list","delete"]
}}
'''

File: pulumi/golang/vault/mini/policies.go
// policies.go
package main

import (
	_ "embed"
	"fmt"
)

//go:embed magistrate.hcl
var Magistrate string

//go:embed template-read.hcl
var TemplateRead string

//go:embed template-admin.hcl
var TemplateAdmin string

func FormatTemplateRead(group string) string {
	return fmt.Sprintf(TemplateRead, group, group)
}

func FormatTemplateAdmin(group string) string {
	return fmt.Sprintf(TemplateAdmin, group, group)
}
